File: src/main/java/com/example/ciidtestspring/entity/Person.java
package com.example.ciidtestspring.entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(unique = true)
    private String email;

    private String phone;

    @ManyToOne
    @JoinColumn(name = "type_id", nullable = false)
    private PersonType type;
}

File: src/app/models/person.ts
export class Person {
    id: number;
    name: string;
    email: string;
    phone: string;
    typeId: number;
  
    constructor(id: number, name: string, email: string, phone: string, typeId: number) {
      this.id = id;
      this.name = name;
      this.email = email;
      this.phone = phone;
      this.typeId = typeId;
    }
  }